src/redux/actions/genes/loadMarkerGenes.js

```diff
@@ -6,26 +6,16 @@ import { fetchWork } from '../../../utils/work/fetchWork';
 import getTimeoutForWorkerTask from '../../../utils/getTimeoutForWorkerTask';
 
 const loadMarkerGenes = (
-  experimentId, resolution, plotUuid, numGenes = 5, selectedCellSets = false,
+  experimentId, resolution, plotUuid, numGenes = 5, selectedCellSets,
 ) => async (dispatch, getState) => {
   // Disabled linter because we are using == to check for both null and undefined values
   // eslint-disable-next-line eqeqeq
   if (experimentId == null || resolution == null) throw new Error('Null or undefined parameter/s for loadMarkerGenes');
 
-  const { experimentSettings } = getState();
-  const { processing } = experimentSettings;
-  const { hierarchy } = getState().cellSets || [];
-  const clusters = hierarchy?.filter((node) => node.key === selectedCellSets)[0]?.children;
-  const { method } = processing.configureEmbedding.clusteringSettings;
   const body = {
     name: 'MarkerHeatmap',
     nGenes: numGenes,
-    type: method,
-    config: {
-      resolution,
-    },
-    typeOfSets: selectedCellSets || 'louvain',
-    clusters,
+    cellSetKey: selectedCellSets || 'louvain',
```

Comment:
why do we need to pass `cellSetKey` at all? What does the worker use this for?

Reply:
The worker use this to access the `cellSet` data inside the `cellSet` object which is keyed by the method name (e.g. `louvain`).

Reply:
Makes sense. Btw, looking at the current state of the code, I think that this line would break if you pass nothing as selectedCellSets (since you are removing the default = false value). But this will be fixed with my suggestion in the comment above.

Reply:
I moved the default value initialization in commit 44ad796 and add a test for it,